src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -423,9 +436,20 @@ private static (ControlInfoJson.Item item, int index) CombineIRAndState(BlockNod
                 resultControlInfo.HasDynamicProperties = hasDynamicProperties;
                 resultControlInfo.AllowAccessToGlobals = templateState?.ComponentManifest?.AllowAccessToGlobals;
             }
-            resultControlInfo.Template = template;
+            resultControlInfo.Template = template.JsonClone();
             resultControlInfo.Children = orderedChildren;
 
+            // Using the stored PCFDynamicSchemaForIRRetrieval/OverridableProperties value for each control instance,
+            // instead of the default value from the control template.
+            if (entropy.OverridablePropertiesEntry.TryGetValue(controlName, out object val))
```

Comment:
nit: Could creating an out object of 'val' here be an issue, or at least redundantt/less-readable? An out object name 'val' already exists above, so we're overloading this variable